src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelModuleResolutionFunction.java

```diff
@@ -366,8 +366,16 @@ static BazelModuleResolutionValue createValue(
     // Calculate a unique name for each used extension id.
     BiMap<String, ModuleExtensionId> extensionUniqueNames = HashBiMap.create();
     for (ModuleExtensionId id : extensionUsagesById.rowKeySet()) {
-      String bestName =
-          id.getBzlFileLabel().getRepository().getName() + "~" + id.getExtensionName();
+      // Ensure that the resulting extension name (and thus the repository names derived from it) do
+      // not start with a tilde.
+      String repoName = id.getBzlFileLabel().getRepository().getName();
+      String nonEmptyRepoPart;
+      if (repoName.isEmpty()) {
```

Comment:
Nit: maybe `id.getBzlFileLabel().getRepository().isMain()` is a bit more clear? Since other Bazel modules should have [a non-empty repo name](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/cmdline/RepositoryName.java;drc=48a8d01b05149757f69a6a65a22a280bf003cd24;bpv=1;bpt=1;l=51)

Reply:
Changed